packages/observable-hooks/src/use-observable-eager-state.ts

```diff
@@ -26,11 +26,14 @@ export function useObservableEagerState<TState>(
   const errorRef = useRef<Error | null>()
   const isAsyncEmissionRef = useRef(false)
 
+  const didSyncEmit = useRef(false)
+
   const [state, setState] = useState<TState>(() => {
     let state: TState
     state$
       .subscribe({
         next: value => {
+          didSyncEmit.current = true
```

Comment:
I did this instead of checking whether `state` was `undefined` because the Observable could genuinely emit an `undefined` value.